client/components/theme-upgrade-modal/index.tsx

```diff
@@ -69,11 +77,15 @@ export const ThemeUpgradeModal = ( {
 	const isDesktop = useBreakpoint( '>782px' );
 
 	// Check current theme: Does it have a plugin bundled?
-	const theme_software_set = theme?.data?.taxonomies?.theme_software_set?.length;
-	const showBundleVersion = theme_software_set;
-	const isExternallyManaged = useSelector( ( state ) => isExternallyManagedTheme( state, slug ) );
+	const themeSoftwareSet = theme?.data?.taxonomies?.theme_software_set as
+		| ThemeSoftwareSet[]
+		| undefined;
+	const showBundleVersion = themeSoftwareSet?.length;
+	const isExternallyManaged = theme?.data?.theme_type === 'managed-external';
 
-	const bundleSettings = useBundleSettings( slug );
+	// Currently, it always get the first software set. In the future, the whole applications can be enhanced to support multiple ones.
+	const firstThemeSoftwareSet = themeSoftwareSet?.[ 0 ];
+	const bundleSettings = useBundleSettings( firstThemeSoftwareSet?.slug );
```

Comment:
Should this be `useBundleSettingsByTheme`?

Reply:
No, as I decided to refrain from using Calypso state this time. 🙂 
https://github.com/Automattic/wp-calypso/pull/85013#discussion_r1421971735

Reply:
CC: @renatho